### PR TITLE
feat(pre-check): adapt to task framework

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-feature
module-ticket
module-task framework

#### What this PR does / why we need it:
This PR makes pre-check task adapt to new task framework.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Self-test passed by PROCESS mode boot.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/runtime/PreCheckTaskParameters.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import com.oceanbase.odc.service.regulation.ruleset.model.Rule;
+
+import lombok.Data;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/1/30 11:06
+ */
+@Data
+public class PreCheckTaskParameters implements Serializable {
```
Comment (krihy) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/runtime/PreCheckTaskParameters.java:36]: implements Serializable is unnecessary

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTaskCopied.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        } catch (VerifyException e) {
+            log.info(e.getMessage());
+        }
+        if (Objects.nonNull(connectionConfig) && needCreatePreCheckTask(taskEntity)) {
+            // Skip SQL pre-check if connection config is null
+            JobDefinition jobDefinition = buildJobDefinition(taskEntity, connectionConfig, riskLevelDescriber);
+            Long jobId = jobScheduler.scheduleJobNow(jobDefinition);
+            taskService.updateJobId(this.preCheckTaskId, jobId);
+            try {
+                // Pre-check task has no timeout yet, set to max value (about 24 days)
```
Comment (krihy) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTaskCopied.java:136]: timeout is to lang, may 1 hours

File: server/integration-test/src/test/java/com/oceanbase/odc/service/objectstorage/LocalFileOperatorTest.java [hunk 3]
```diff
@@ -44,7 +44,7 @@ public class LocalFileOperatorTest extends ServiceTestEnv {
 
     private static final String BUCKET = "test-bucket";
 
-    @InjectMocks
+    @Autowired
```
[the remark this answers is not in the document]
Comment (smallsheeeep) [reply server/integration-test/src/test/java/com/oceanbase/odc/service/objectstorage/LocalFileOperatorTest.java:47]: But the previous codes could not work when I change the constructor. Using `@Autowired` seems right here.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/runtime/PreCheckTaskParameters.java (hunk 1 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/runtime/PreCheckTaskParameters.java:36]: This is done to be consistent with the style of the existing task parameters, and shouldn't introduce any additional disadvantages.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTaskCopied.java (hunk 2 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/PreCheckRuntimeFlowableTaskCopied.java:136]: Yes you are right, I add a system config to describe it and the default vulue is 1 hour.

**Review by krihy — APPROVED** [review]
LGTM

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**